feat: add CLI flags support to attach command

simondmorias opened the pull request:
## Summary

Add support for `--model`, `--session`, `--continue`, `--prompt`, `--agent`, `--title`, and `--command` flags to the `opencode attach` command, bringing feature parity with the main TUI and run commands.

## Changes

- **Added CLI flags to AttachCommand builder**:
  - `--model` / `-m` - Model to use (format: provider/model)
  - `--session` / `-s` - Session ID to continue
  - `--continue` / `-c` - Continue the last session
  - `--prompt` / `-p` - Prompt to use
  - `--agent` - Agent to use
  - `--title` - Title for the session
  - `--command` - Command to run (uses prompt for args)

- **Updated Args interface** to include `title` and `command` properties

- **Implemented stdin piping support** for prompts in attach command

- **Added command pre-fill support** in home route - prepends `/` to command name and combines with prompt

- **Updated documentation** with complete flag reference and usage examples

## Usage Examples

```bash
# Attach with specific model and agent
opencode attach http://localhost:4096 --model anthropic/claude-3-5-sonnet --agent build

# Attach and continue last session
opencode attach http://localhost:4096 --continue

# Attach with pre-filled command
opencode attach http://localhost:4096 --command commit --prompt "with conventional format"

# Attach and provide initial prompt
echo "Review this code" | opencode attach http://localhost:4096 --prompt "and suggest improvements"

# Attach to specific session with title
opencode attach http://localhost:4096 --session abc123 --title "Bug fix session"
```

## Files Changed

- `packages/opencode/src/cli/cmd/tui/attach.ts` - Added flag definitions and argument passing
- `packages/opencode/src/cli/cmd/tui/context/args.tsx` - Extended Args interface
- `packages/opencode/src/cli/cmd/tui/routes/home.tsx` - Added command pre-fill logic
- `packages/web/src/content/docs/cli.mdx` - Added attach command documentation

## Testing

- No new TypeScript errors introduced (verified with `bun typecheck`)
- All flags follow the same patterns as the `thread` command for consistency
- Documentation includes examples and complete flag reference
- Manual test: `bun run --cwd packages/opencode --conditions=browser src/index.ts \
  attach http://localhost:4096 --agent plan`

## Notes

- The `--format` flag was intentionally not added as it's only relevant for non-interactive output (like the `run` command), not for the TUI
- Pre-existing TypeScript errors in `script/build.ts` and `src/mcp/index.ts` are unrelated to these changes

simondmorias committed:
feat: add CLI flags support to attach command

Add support for --model, --session, --continue, --prompt, --agent, --title, and --command flags to the 'opencode attach' command, bringing feature parity with the main TUI and run commands.

Changes:
- Added all missing CLI flags to AttachCommand builder
- Updated Args interface to include title and command properties
- Implemented stdin piping support for prompts
- Added command pre-fill support (prepends / to command name)
- Updated documentation with complete flag reference and examples

This allows users to attach to a running server with pre-configured settings like:
  opencode attach http://localhost:4096 --model anthropic/claude-3-5-sonnet --agent build
  opencode attach http://localhost:4096 --command commit --prompt 'with conventional format'

rekram1-node committed:
Merge branch 'dev' into feat-attach-flags

rekram1-node commented:
/review

simondmorias committed:
Update packages/opencode/src/cli/cmd/tui/routes/home.tsx

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

simondmorias committed:
Merge branch 'dev' into feat-attach-flags

simondmorias committed:
fix: remove duplicate lines in home.tsx

rekram1-node commented:
/review

rekram1-node commented on packages/opencode/src/cli/cmd/tui/context/args.tsx:
```diff
@@ -6,6 +6,8 @@ export interface Args {
   continue?: boolean
   sessionID?: string
+  title?: string
+  command?: string
```
The title flag doesnt really make sense here?
Neither does command....


All the other ones do

simondmorias committed:
fix: remove title and command flags from attach command

These flags don't make sense in the TUI context:
- title: was never used in the TUI flow
- command: pre-filling slash commands is unusual for interactive TUI

Also resolves merge conflict with upstream dev branch.

simondmorias commented:
Agreed - removed `title` and `command` flags. They don't make sense in the TUI context:
- `title`: was added to the Args interface but never actually used in the TUI flow
- `command`: pre-filling slash commands is unusual for an interactive TUI where you'd typically just type the command yourself

Also resolved merge conflicts with upstream dev.
